engine/src/lib.rs

```diff
@@ -120,7 +124,18 @@ mod contract {
         }
 
         let args: NewCallArgs = io.read_input_borsh().sdk_unwrap();
-        state::set_state(&mut io, args.into()).sdk_unwrap();
+        let state: EngineState = args.into();
+
+        let blockchain_hashchain = BlockchainHashchain::new(
+            &state.chain_id,
+            b"aurora",
+            io.block_height(),
+            [0; 32],
+            [0; 32],
+        );
+
+        state::set_state(&mut io, state).sdk_unwrap();
+        hashchain::set_state(&mut io, blockchain_hashchain).sdk_unwrap();
```

Comment:
Adding an extra entry to the state in the contract only will make many tests fail because the standalone engine will not produce the same state. It's fine for now while this PR is in draft and we are mostly just trying to measure the gas cost of this change, but we will eventually need to fix it so that the standalone engine is capturing these state changes too.

Reply:
This is done. The standalone engine is now on sync with the contract.